### ISSUE TITLE
Code generated for a mutation containing a union is incorrect

### ISSUE THREAD

**Issue by ChrisMH** [ChilliCream/hotchocolate#3546]
Code generated for a mutation containing a union is incorrect

preview.27

For this mutation:
```
mutation StoreUserSettingFor($userId: Int!, $customerId: Int!, $input: StoreUserSettingForInput!) {
    storeUserSettingFor(userId: $userId, customerId: $customerId, input: $input) {
        ... on UserSettingSuccess {id}
    }
}
```

SS is generating this code:
```
        private global::GraphTest.Graphs.IStoreUserSettingFor_StoreUserSettingFor MapNonNullableIStoreUserSettingFor_StoreUserSettingFor(global::GraphTest.Graphs.State.IUserSettingResultData data)
        {
            IStoreUserSettingFor_StoreUserSettingFor? returnValue;
            if (data is global::GraphTest.Graphs.State.UserSettingSuccessData userSettingSuccess)
            {
                if (userSettingSuccess.Id.HasValue)
                {
                    throw new global::System.ArgumentNullException();
                }

                returnValue = new global::GraphTest.Graphs.StoreUserSettingFor_StoreUserSettingFor_UserSettingSuccess(userSettingSuccess.Id!.Value);
            }
            else if (data is global::GraphTest.Graphs.State.UserSettingErrorData userSettingError)
            {
                returnValue = new global::GraphTest.Graphs.StoreUserSettingFor_StoreUserSettingFor_UserSettingError();
            }
            else
            {
                throw new global::System.NotSupportedException();
            }

            return returnValue;
        }
```

When calling the mutation, I am getting an ArgumentNullException from line 5 of the above function.  I believe that the check on line 4 should be `if (userSettingSuccess.Id.HasValue)` (note the !)

**Comment by freever** [comment]
I have been getting this issue too

![image](https://user-images.githubusercontent.com/2269624/115344219-5a388600-a1ad-11eb-8b32-a12e70485eac.png)

**Comment by PascalSenn** [comment]
@freever  This should be resolved in  #3549 What version are you on?

**Comment by freever** [comment]
@PascalSenn Sorry, forgot to mention this is on 11.2.0-preview.27

**Comment by PascalSenn** [comment]
@freever  Can you try 11.2.0?

**Comment by freever** [comment]
I was having a different problem with 11.2.0 which blocks me from testing this
https://github.com/ChilliCream/hotchocolate/issues/3544

**Comment by ChrisMH** [comment]
The original issue is fixed for me in 11.2.0

**Comment by freever** [comment]
@PascalSenn yes it's fixed for me too in 11.2.0 and my other problem seems to have disappeared
